src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -0,0 +1,229 @@
+      const isIncognito = type === 'incognito';
+      await followTopic(topicName, isIncognito);
+    } catch (error) {
+      if (__DEV__) {
+        console.log(error);
+      }
+    }
+  };
+
+  return (
+    <View>
+      <BottomSheetV2
+        ref={ref}
+        closeOnPressMask={true}
+        height={290}
+        keyboardAvoidingViewEnabled={true}
+        viewstyle={{
+          paddingHorizontal: 0,
+          paddingTop: 0
+        }}>
+        <View style={styles.containerBottomSheet}>
+          <View style={styles.titleContainer}>
+            <View
+              style={{
+                width: 12,
+                height: 12
+              }}
+            />
+            <Text style={styles.title}>Join as</Text>
+
+            <TouchableOpacity onPress={onClose}>
+              <IconClose color={COLORS.black} />
+            </TouchableOpacity>
+          </View>
+          <View
+            style={{
+              borderTopColor: COLORS.gray,
+              borderTopWidth: 1
+            }}
+          />
+          <UserItem
+            type="signed"
+            username={props.username}
+            profilePicture={props.profilePicture}
+            isFollowing={followType === 'signed'}
+            onPress={() => {
+              handleFollowTopic({
+                type: 'signed'
+              });
+            }}
+          />
+          <UserItem
+            type="incognito"
+            username="Incognito"
+            profilePicture=""
+            isFollowing={followType === 'incognito'}
+            onPress={() => {
+              handleFollowTopic({
+                type: 'incognito'
+              });
+            }}
+          />
+          <View
+            style={{
+              paddingVertical: dimen.normalizeDimen(14),
+              paddingHorizontal: dimen.normalizeDimen(20)
+            }}>
+            <Button
+              onPress={() => {
+                handleUnfollowTopic({
+                  type: followType
+                });
+              }}
+              disabled={followType === ''}
+              styles={{
+                backgroundColor: COLORS.white,
+                paddingVertical: dimen.normalizeDimen(11),
+                paddingHorizontal: dimen.normalizeDimen(20),
+                borderRadius: dimen.normalizeDimen(8),
+                height: dimen.normalizeDimen(40),
+                flexDirection: 'row',
+                justifyContent: 'center',
+                alignItems: 'center',
+                borderColor: isFollow ? COLORS.redalert : COLORS.light_silver,
+                borderWidth: 1
+              }}
+              textStyling={{
+                fontFamily: fonts.inter[500],
+                fontWeight: 'normal',
+                fontSize: 12,
+                color: isFollow ? COLORS.redalert : COLORS.white
+              }}>
+              Leave Community
+            </Button>
+          </View>
+        </View>
+      </BottomSheetV2>
+    </View>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The component returned by `BottomSheetFollow` seems quite large and complex. Consider breaking it down into smaller, reusable components to improve readability and maintainability. For example, you could create separate components for the UserItems and the Button.

```diff
-  return (
-    <View>
-      ...
-    </View>
-  );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->